CMakeLists.txt

```diff
@@ -1,77 +1,96 @@
+    if (CONFIG_LV_TICK_CUSTOM)
+        if (CONFIG_LV_TICK_CUSTOM_SYS_TIME_EXPR)
+            target_compile_definitions(${COMPONENT_LIB} PUBLIC "-DLV_TICK_CUSTOM_SYS_TIME_EXPR=${CONFIG_LV_TICK_CUSTOM_SYS_TIME_EXPR}")
+        endif()
     endif()
-endif()
 
-if (CONFIG_LV_TICK_CUSTOM)
-    if (CONFIG_LV_TICK_CUSTOM_SYS_TIME_EXPR)
-        target_compile_definitions(${COMPONENT_LIB} PUBLIC "-DLV_TICK_CUSTOM_SYS_TIME_EXPR=${CONFIG_LV_TICK_CUSTOM_SYS_TIME_EXPR}")
+    if (CONFIG_LV_USER_DATA_FREE)
+        target_compile_definitions(${COMPONENT_LIB} PUBLIC "-DLV_USER_DATA_FREE=${CONFIG_LV_USER_DATA_FREE}")
     endif()
-endif()
 
-if (CONFIG_LV_USER_DATA_FREE)
-    target_compile_definitions(${COMPONENT_LIB} PUBLIC "-DLV_USER_DATA_FREE=${CONFIG_LV_USER_DATA_FREE}")
-endif()
+    if (CONFIG_LV_ATTRIBUTE_FAST_MEM_USE_IRAM)
+        target_compile_definitions(${COMPONENT_LIB} PUBLIC "-DLV_ATTRIBUTE_FAST_MEM=IRAM_ATTR")
+    endif()
 
-if (CONFIG_LV_ATTRIBUTE_FAST_MEM_USE_IRAM)
-    target_compile_definitions(${COMPONENT_LIB} PUBLIC "-DLV_ATTRIBUTE_FAST_MEM=IRAM_ATTR")
-endif()
+    elseif(ZEPHYR_BASE)
 
-elseif(ZEPHYR_BASE)
+    if(CONFIG_LVGL)
 
-if(CONFIG_LVGL)
+        zephyr_include_directories(${ZEPHYR_BASE}/lib/gui/lvgl)
 
-zephyr_include_directories(${ZEPHYR_BASE}/lib/gui/lvgl)
+        target_include_directories(lvgl INTERFACE ${CMAKE_CURRENT_SOURCE_DIR})
 
-target_include_directories(lvgl INTERFACE ${CMAKE_CURRENT_SOURCE_DIR})
+        zephyr_compile_definitions(LV_CONF_KCONFIG_EXTERNAL_INCLUDE=<autoconf.h>)
 
-zephyr_compile_definitions(LV_CONF_KCONFIG_EXTERNAL_INCLUDE=<autoconf.h>)
+        zephyr_compile_definitions_ifdef(CONFIG_LV_MEM_CUSTOM
+            LV_MEM_CUSTOM_ALLOC=${CONFIG_LV_MEM_CUSTOM_ALLOC}
+            )
+        zephyr_compile_definitions_ifdef(CONFIG_LV_MEM_CUSTOM
+            LV_MEM_CUSTOM_FREE=${CONFIG_LV_MEM_CUSTOM_FREE}
+            )
+        zephyr_compile_definitions_ifdef(CONFIG_LV_TICK_CUSTOM
+            LV_TICK_CUSTOM_SYS_TIME_EXPR=${CONFIG_LV_TICK_CUSTOM_SYS_TIME_EXPR}
+            )
 
-zephyr_compile_definitions_ifdef(CONFIG_LV_MEM_CUSTOM
-    LV_MEM_CUSTOM_ALLOC=${CONFIG_LV_MEM_CUSTOM_ALLOC}
-    )
-zephyr_compile_definitions_ifdef(CONFIG_LV_MEM_CUSTOM
-    LV_MEM_CUSTOM_FREE=${CONFIG_LV_MEM_CUSTOM_FREE}
-    )
-zephyr_compile_definitions_ifdef(CONFIG_LV_TICK_CUSTOM
-    LV_TICK_CUSTOM_SYS_TIME_EXPR=${CONFIG_LV_TICK_CUSTOM_SYS_TIME_EXPR}
-    )
+        zephyr_library()
 
-zephyr_library()
+        file(GLOB_RECURSE SOURCES src/*.c)
+        zephyr_library_sources(${SOURCES})
 
-file(GLOB_RECURSE SOURCES src/*.c)
-zephyr_library_sources(${SOURCES})
+    endif(CONFIG_LVGL)
 
-endif() # CONFIG_LVGL
+else(ESP_PLATFORM)
 
-else()
+    file(GLOB_RECURSE SOURCES src/*.c)
+    file(GLOB_RECURSE EXAMPLE_SOURCES ${CMAKE_CURRENT_LIST_DIR}/examples/*.c)
 
-file(GLOB_RECURSE SOURCES ${CMAKE_CURRENT_LIST_DIR}/src/*.c)
-add_library(lvgl STATIC ${SOURCES})
+    if(MICROPY_DIR)
```

Comment:
ESP32+Micropython already use LVGL as a library, and this change would break it.  

Your changes are specific to the RPi port, and [as discussed](https://github.com/lvgl/lv_micropython/pull/42#discussion_r700865385) I think it's best to look for a way to do something similar to the ESP32 here instead of adding LVGL sources directly to Micropython.